extensions/shared/upgrade-nudge/store.js

```diff
@@ -40,7 +39,11 @@ registerStore( 'wordpress-com/plans', {
 
 	controls: {
 		FETCH_FROM_API( { url } ) {
-			return apiFetch( { url, credentials: 'omit', mode: 'no-cors' } );
+			// We cannot use `@wordpress/api-fetch` here since it unconditionally sends
+			// the `X-WP-Nonce` header, which is disallowed by WordPress.com
+			return fetch( url, { credentials: 'same-origin', mode: 'cors' } ).then( response =>
+				response.json()
+			);
```

Comment:
This header is sent per registration of a particular middleware, defined in [`createNonceMiddleware`](https://github.com/WordPress/gutenberg/blob/08cda3f0dd23595e685d67209e4b4f811d2f824d/packages/api-fetch/src/middlewares/nonce.js), and whose addition is performed by a WordPress inline script:

https://github.com/WordPress/gutenberg/blob/08cda3f0dd23595e685d67209e4b4f811d2f824d/lib/client-assets.php#L246-L270

I say this as an outsider, but would there be benefits to swapping that core hook with a Jetpack-ready one?

Reply:
:wave: @mcsf

> I say this as an outsider, but would there be benefits to swapping that core hook with a Jetpack-ready one?

I'm not sure that would help TBH, since the middleware [will still send an `X-WP-Nonce` header](https://github.com/WordPress/gutenberg/blob/08cda3f0dd23595e685d67209e4b4f811d2f824d/packages/api-fetch/src/middlewares/nonce.js#L17), albeit a customized one, and the problem with WP.com seems to be that it hates that header regardless:

> Request header field x-wp-nonce is not allowed by Access-Control-Allow-Headers in preflight response.

(Hope I'm reading your suggestion correctly.)

Reply:
My initial suggestion was to prevent that core middleware from being registered at all, and replacing it with something similar but JP-friendly. That said, a safer and more maintainable way to achieve the same would be to register another piece of middleware, in JP, that unsets the core middleware's nonce header.

> the problem with WP.com seems to be that it hates that header regardless

I was working around that aspect, but it begs the question: why does WP.com hate the header, and can/should we change that? If not, my middleware-based suggestion still stands.

Reply:
> My initial suggestion was to prevent that core middleware from being registered at all, and replacing it with something similar but JP-friendly. That said, a safer and more maintainable way to achieve the same would be to register another piece of middleware, in JP, that unsets the core middleware's nonce header.
> 
> > the problem with WP.com seems to be that it hates that header regardless
> 
> I was working around that aspect, but it begs the question: why does WP.com hate the header, and can/should we change that? If not, my middleware-based suggestion still stands.

I agree it's a good question to ask, and I'm willing to look further into this (but wouldn't want it to block this PR).

For the purpose of this PR: Given the choice between

- `fetch`, which comes with the right defaults, and doesn't set the `X-WP-Nonce` header, and
- `apiFetch`, which sets different defaults (`credentials: include` specifically) that I need to override, and which adds the header through a middleware, which I'd need to prevent from being registered, or to override by adding another middleware :sweat_smile: 

... I kinda have a clear preference, TBH :grimacing: _Non sunt multiplicanda entia praeter necessitatem_

Reply:
Let's rewrite our acronym-based _adagia_ into Latin—I love it!

As for the matter: now, that's totally fair, and in its current iteration the PR isolates `fetch` quite nicely. I think it's still pertinent to circle back to this and figure out whether there's room for `apiFetch` to grow: is Jetpack's case generalisable to other _entibus_? That work doesn't have to block this!